emcc.py

```diff
@@ -2370,9 +2377,9 @@ def consume_arg():
       newargs[i + 1] = ''
```

Comment:
I think the line before this (`newargs[i] = ''`) can be replaced with an assertion, as we should always call `check_flag` first to zero out the flag, and then `consume_arg` only consumes the argument after it, but does not need to modify the thing before the arg? That's more consistent with the name too.

Reply:
Agreed.  Although this is existing code, so you are essentially suggesting and additional separate refactor here.   I'm planning on a followup which will allow `--foo=bar` as well as `--foo bar` and I think this will get rolled into that change.